guitar-api.yaml

```diff
@@ -0,0 +1,122 @@
+openapi: 3.0.3
+info:
+  title: Guitar API
+  description: |-
+   Reference Basic infromation on guitar models and the musicians that have used them at various points in their careers.
+    - [Guitar API Repository](https://github.com/Bobs-code/Guitar-API)
+  contact:
+    email: robertgriffintm@gmail.com
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+servers:
+  - url: http://localhost:8080
+tags:
+  - name: guitar
+    description: Everything about popular guitar models
+    externalDocs:
+      description: Find out more
+      url: http://swagger.io
+paths:
+  /guitar:
+    get:
+      tags: 
+        - guitar
+      summary: find a guitar by ID
+      description: returns a single guitar
+      operationId: getGuitarByID
+      parameters:
+        - name: guitarId
+          in: query
+          schema:
+            type: integer
+          required: true
+          description: Id of guitar to return
+      responses:
+        200:
+          description: returned requested guitar
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/guitar'
+    post:
+      tags: 
+        - guitar
+      summary: add guitar
+      description: add a guitar to the catalog
+      operationId: addGuitar
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/guitar'
+      responses:
+        201:
+          description: Guitar successfully added
+    delete:
+      tags: 
+          - guitar
+      summary: delete a guitar
+      parameters:
+        - in: query
+          name: id
+          schema:
+            type: integer
+          required: true
+          description: Id of guitar to be deleted
+      responses:
+        '204':
+          description: Guitar successfully deleted
+        '400':
+          description: Invalid ID provided
+        '500':
+          description: Internal server error
+    put:
```

Comment:
```suggestion
    patch:
```

Nitpick: This is also a convention thing—PATCH is generally meant for updating, especially since you have a POST.

For all practical purposes, POST and PUT are functionally the same (they have some differences, but I don’t remember off the top of my head and have to look it up). You usually won’t see both side-by-side and will encounter only one in a schema (usually). With a slight preference for PUT.

So to summarize, I’d change:

- **Create**: PUT
- **Update**: PATCH

Again, just purely out of predictable convention that you wouldn’t have known, and is more of a guideline than a hard rule.

Reply:
**Extra info**: shooting _far_ off into the weeds, I’d generally expect PUT to be a more clear-cut “this is **creating** a resource” (what you’re doing).

POST is more of an ambiguous, complex “there’s a lot of stuff going on here that’s hard to explain” operation that you probably shouldn’t have in your guitar API project.

Reply:
Great. I'll get that sorted out too :)